lib/deadletterProcessors/reenqueueByName.js

```diff
@@ -0,0 +1,12 @@
+const deadletterActions = require('./../deadletterActions');
+const Q = require('q');
+
+module.exports = function(event, message) {
+  const content = JSON.parse(message.content);
+
+  if (event === content.event) {
+    return Q.when(deadletterActions.REENQUEUE_MESSAGE);
+  } else {
+    return Q.reject(new Error(`Required ${event} does not match ${message.content.event}`));
```

Comment:
not sure we need `Q` for these, should be able to use native promises

Reply:
while I agree with that and I'd use native as much as possible, but I'm not sure (maybe I'm unaware) of a `Q.when()` implementation in the Promise prototype.

Reply:
in this case the parameter is just a string so `Promise.resolve()` could work? i find the `when` method mysterious indeed

Reply:
@earobinson @iain8 - I've created an issue for this https://github.com/hubba/magic-service-bus/issues/10 in order to properly investigate this.

since the `Q` implementation is in step with the rest of the library, I think investigating it is definitely beneficial to reduce unnecessary dependancies.